### ISSUE TITLE
How to build on Windows

### ISSUE THREAD

**Issue by CodingSpiderFox** [jgaa/restc-cpp#8]
I am running into multiple problems:

a) My depencies (boost, OpenSSL, zlib, etc.) are not installed under C:\devel\...
I install my dependencies using the conan package manager. I changed the paths in the CMake files you provided with the source code.
I also included the conanbuildinfo.cmake file which is generated by conan.

That way I could create a solution for Visual Studio 2015 using CMake.
But I cannot build it. It complains about the libs from boost being missing: LNK1104	cannot open file 'libboost_log-vc140-mt-1_63.lib'

In the Windows ecosystem, the boost libraries are build with these names following this file name pattern: "boost_log-vc140-mt-1_63.lib"

I want to try your lib. Can you please help me getting my workspace to build?

**Comment by jgaa** [comment]
I'll investigate it.

**Comment by CodingSpiderFox** [comment]
Thank you very much in advance

This is how I'm trying to build it at the moment:
I created the conanfile.txt you can see in the repo, containing the dependcies list.
Then I changed the CMakeLists.txt (see commit e1213065d033e31fb89873cc9cabd5d897d63d5b)
What I did there is remove the find_package macro calls as they don't work
Instead I included the conanbuildinfo.cmake (will be generated) file at the top and call "conan_basic_setup()"
I also changed the paths of OpenSSL and zlib to be on the safe side. They point to the source code

I do the basic steps
git clone https://github.com/CodingSpiderFox/restc-cpp
cd restc-cpp
git submodule init
git submodule update

Then I install the conan dependencies (conan can be installed via python "pip install conan")

conan install . --build Boost --build zlib --build OpenSSL -s arch=x86_64 -s build_type=Release -s os=Windows -s compiler="Visual Studio" -s compiler.runtime=MD -s compiler.version=14
This can take a while

cd externals/unittest-cpp/
mkdir build
cd build
cmake .. -G "Visual Studio 14 2015 Win64"
cmake --build . --config Release

cd ../../..

mkdir dbuild
cd dbuild
cmake .. -G "Visual Studio 14 2015 Win64"
cmake --build . --config Release

Leads to many unresolved externals

e.g. 
LINK : fatal error LNK1181: cannot open input file 'UnitTest++.lib' [C:\Users\user\Documents\Projekte\Coding\myrestc-c
pp\restc-cpp\dbuild\tests\async_sleep_tests.vcxproj]
restc-cpp.lib(ConnectionPoolImpl.obj) : error LNK2019: unresolved external symbol SSL_get_shutdown referenced in func
tion "public: class boost::system::error_code const & __cdecl boost::asio::ssl::detail::engine::map_error_code(class bo
ost::system::error_code &)const " (?map_error_code@engine@detail@ssl@asio@boost@@QEBAAEBVerror_code@system@5@AEAV675@@Z
) [C:\Users\user\Documents\Projekte\Coding\myrestc-cpp\restc-cpp\dbuild\tests\readme_tests.vcxproj]

**Comment by CodingSpiderFox** [comment]
Can you tell me when you will have time to look into it? 
Then I can decide better what I will do next on my project

**Comment by jgaa** [comment]
I will look at it Tuesday next week. I have to install a fresh Windows VM to have a clean test environment.

**Comment by CodingSpiderFox** [comment]
Thanks a lot already for letting me know!

**Comment by jgaa** [comment]
I'm currently stalled, waiting for Microsoft to activate a Visual Studio / MSDN subscription. I started this process last Friday, and it seems to be very hard for Microsoft to figure out how to do this.

**Comment by CodingSpiderFox** [comment]
Maybe you can use the community version in the meantime and then upgrade it ;)

**Comment by jgaa** [comment]
I prefer to trouble shoot on a fresh install. That way I don't break anything on my windows development machine, and I get a predictable test environment. Hopefully, Microsoft gets this resolved this week.

**Comment by CodingSpiderFox** [comment]
OK

**Comment by CodingSpiderFox** [comment]
Did you get your activation meanwhile?

**Comment by jgaa** [comment]
I finally got msdn working today (it took 12 working hours over 2 1/2 weeks, involving 4 MS support engineers from two Microsoft divisions). It's late in Europe now, so I will start working on this issue tomorrow morning.

**Comment by CodingSpiderFox** [comment]
OK, thank you

**Comment by jgaa** [comment]
I have looked at conan. I like the concept. The best (simplest) way to support this kind of build is to add knowledge of conan in my cmake files, and to pick the location of the dependencies automatically. I'm a bit concerned about the overhead in regards of testing in the future (it doubles the number of builds). However, it will make life a lot simpler for occasional developers who want to play with the library.

I'll experiment a little more with conan and read trough the documentation tomorrow. (There are some potential issues. I was unable to build openssl using Python 3. It seemed to get it from a repository when I did not specify the build option, but I found no binary files.)

I guess I will have something running by Friday afternoonn (Eastern European Time Zone).

**Comment by jgaa** [comment]
I have detached my habits regarding local libraries from the cmake file, and provided a build script that shows how to build the project when you have locally built versions of zlib, openssl and boost.

I have also played around a bit with conan, and provided a script and documentation on how to build the project with conan. I did however find some issues. At this time it's not clear to me exactly what is wrong. For example, debug build does not link at all with boost, of boost is built dynamically.
